Salwador/UIComponents/SettingsView/SettingsView.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  SettingsView.swift
+//  Salwador
+//
+//  Created by Ailiniyazi Maimaiti on 07.03.23.
+//
+
+import SwiftUI
+
+struct SettingsView: View {
+    @Binding var darkModeEnabled: Bool
+    @Binding var systemSchemaEnabled: Bool
+
+    var body: some View {
+        NavigationView {
+            Form {
+                Section(header: Text("Settings")) {
+                    Toggle(isOn: $darkModeEnabled) {
+                        Text("Dark Mode:")
+                    }
+                    .onChange(of: darkModeEnabled, perform: { _ in
+                        ThemeManager.shared.handleTheme(darkMode: darkModeEnabled, systemSchema: systemSchemaEnabled)
+
+                    })
+
+                    Toggle(isOn: $systemSchemaEnabled) {
+                        Text("Use System Schema:")
+                    }
+                    .onChange(of: systemSchemaEnabled, perform: { _ in
+                        ThemeManager.shared.handleTheme(darkMode: darkModeEnabled, systemSchema: systemSchemaEnabled)
+
+                    })
```

Comment:
idea: So when i look at the data flow for those two simple booleans i would suggest a different approach. 

What do we have: 
- AppStorage property so save data into user defaults
- Theme manager that can set dark mode/ light mode / system preset

I think it's not great that we read the data out of AppStorage, pass it as binding to views and then react to value change of them with a passing to the ThemeManager to set.  

I suggest to not have the AppStorage values where they are not needed and since we need the ThemeManager one possible solution could be 

```swift
public enum UIMode: Int {
    case system
    case light
    case dark

    public var interfaceStyle: UIUserInterfaceStyle {
        switch self {
        case .system:
            return .unspecified
        case .light:
            return .light
        case .dark:
            return .dark
        }
    }
}

class ThemeManager {
    @AppStorage("uiModeSet") private var uiModeSet: UIMode = .system

    static let shared = ThemeManager()

    private init() {}

    func setMode(to newMode: UIMode) {
        UIApplication.shared.windows.first?.overrideUserInterfaceStyle = newMode.interfaceStyle
    }

    func restoreMode() {
        UIApplication.shared.windows.first?.overrideUserInterfaceStyle = uiModeSet.interfaceStyle
    }
}
```

and then do a 

```swift
.onAppear {
    ThemeManager.shared.restoreMode()
}
```
in ```AppNavigationView``` 

Additionally a different way of setting the value in ThemeManager than those bindings used in SettingsView is needed.